operator/pkg/client/cluster.go

```diff
@@ -40,6 +46,22 @@ func (c *Factory) redpandaAdminForCluster(cluster *redpandav1alpha2.Redpanda) (*
 	return client, nil
 }
 
+func (c *Factory) redpandaAdminForV1Cluster(cluster *vectorizedv1alpha1.Cluster) (*rpadmin.AdminAPI, error) {
+	ctx := context.Background()
+
+	// Assume no TLS. Practically, we don't need to support it in Operator V1.
+	t := &certmanager.ClusterCertificates{}
+
+	a, err := admin.NewNodePoolInternalAdminAPI(ctx, c.Client, cluster, fmt.Sprintf("%s.%s.svc.cluster.local", cluster.Name, cluster.Namespace), t)
```

Comment:
Not sure if you want to try testing this outside of `kuttl`, but heads up that there's the whole "dialer" field attached to the factory predominantly for testing purposes. What we do in our go-based tests is pass in a "`PodDialer`" that basically does some kubeproxy hole-punching so that we can dial into an unexposed port from our host machine. May want to consider eventually adding support for some stuff like that if you want to migrate off of `kuttl` like we do.

Other thing to note, this is great in general! If we have implementations initializing schema registry clients and kafka clients similar to what you've added here, then it should actually be pretty trivial to support declarative user/schema/acl management in v1.